lib/mayaUsd/ufe/UsdUndoRenameCommand.cpp

```diff
@@ -131,8 +133,13 @@ bool UsdUndoRenameCommand::renameRedo()
         _newName = uniqueName(childrenNames, _newName);
     }
 
+    // names are not allowed to start to digit numbers
+    if(std::isdigit(_newName.at(0))){
+        _newName = prim.GetName();
+    }
```

Comment:
Names are not allowed to start with digits and should to replaced with the old name. This is based on Natalia's request to match Maya's behavior.

Reply:
Hmm, Maya's behavior is confusing / buggy.  Here's what I get when I rename a sphere with a leading numeric:
select -r pSphere1 ;
rename "pSphere1" "1pSphere1";
// Warning: line 1: New name contains invalid characters. Illegal characters were converted to "_". //
// Result: pSphere1 //

Yes, you do get the old name back, but you also get a warning saying "illegal character" will be replaced with an underscore --- and then you don't get the underscore...

Reply:
> 
> 
> Hmm, Maya's behavior is confusing / buggy. Here's what I get when I rename a sphere with a leading numeric:
> select -r pSphere1 ;
> rename "pSphere1" "1pSphere1";
> // Warning: line 1: New name contains invalid characters. Illegal characters were converted to "_". //
> // Result: pSphere1 //
> 
> Yes, you do get the old name back, but you also get a warning saying "illegal character" will be replaced with an underscore --- and then you don't get the underscore...

@ppt-adsk Can't agree more. I have noticed this as well and it is very confusing. 

I've already brought this up with design team and not really sure if we want to mimic exactly how Maya's handling all this cases. 

My main goal for now is to prevent Maya crash when illegal characters are used in names.

Reply:
O.K., fair enough, so for now, no crash, but no warning, and no underscore.  Works for me.